contrib/src/main/java/com/datatorrent/contrib/hbase/HBaseWindowStore.java

```diff
@@ -132,8 +140,11 @@ public void beginTransaction() {
 
   @Override
   public void commitTransaction() {
-    // HBase does support transactions so this method left empty
-
+    try {
+      flushTables();
```

Comment:
The reason for doing this as part of commitTransaction is that we are piggybacking on the mechanism to minimize rewrites on a replay from failure. Please see the long description above on the workings of these operators. There are existing common store operators that implement the fault tolerance mechanism by using the transaction pattern. If transactions are supported it can result in exactly once otherwise it will help in reducing writing duplicates significantly.